### DISCUSSION TITLE
Pin a document to a specific ReSpec version

### DISCUSSION THREAD

**Discussion by ktk** [speced/respec#4445] [General]
I see there are releases of ReSpec on Github and they seem to get pushed to NPM as well. Is there a way to pin to a specific version instead of latest in a document so we don't break stuff without explicitly asking for it?

**Comment by sidvishnoi** [comment]
Given it's on npm:
1. You can use CDN like unpkg (and many others) to pin specific version. For example:
   ```
   https://unpkg.com/respec@34.1.1/builds/respec-w3c.js
   https://unpkg.com/respec@34/builds/respec-w3c.js
   ```
2. Self-host it: Copy the `respec-w3c` script and push it your repository. 
    1. You can simplify updates with self-hosting with using a package.json and excluding the respec script from `.gitignore`.
    2. Or, you do something more automated like MediaCapture spec does: https://github.com/w3c/mediacapture-main/blob/main/.github/workflows/update-respec.yml

**Comment by marcoscaceres** [comment]
This is **strongly** discouraged. Doing this mean you will potentially end up with the wrong templates, the wrong W3C Process checks, and generally in a world of pain when you move to a new version. 

If Repec is yelling at you about things and a spec is broken, then it's doing its job correctly: it means that the spec needs to be updated and maintained. Locking to a specific version of Respec means that the spec is not being maintained properly.
